atoma-vllm/src/llm_service.rs

```diff
@@ -45,40 +61,69 @@ pub struct LlmService {
     request_counter: u64,
     /// A request validation instance
     validation_service: Validation,
+    /// Tokenizer handle
+    tokenizer_handle: JoinHandle<Result<(), LlmServiceError>>,
+    /// Shutdown signal
+    shutdown_signal: mpsc::Receiver<()>,
+    /// Tracing span
+    span: Span,
 }
 
 impl LlmService {
     /// Starts the service
     #[instrument(skip_all)]
     #[allow(clippy::too_many_arguments)]
-    pub async fn start<M>(
+    pub async fn start<M, T: AsRef<Path>>(
         api_key: String,
         atoma_event_subscriber_receiver: UnboundedReceiver<GenerateRequest>,
         atoma_client_sender: UnboundedSender<Vec<GenerateRequestOutput>>,
         cache_config: CacheConfig,
-        cache_dir: PathBuf,
+        cache_dir: T,
         device: Device,
         dtype: DType,
         flush_storage: bool,
         model_name: String,
+        num_tokenizer_workers: usize,
         revision: String,
         scheduler_config: SchedulerConfig,
-        tokenizer: Tokenizer,
+        tokenizer_receiver: mpsc::UnboundedReceiver<EncodeTokenizerRequest>,
         validation_service: Validation,
+        shutdown_signal: mpsc::Receiver<()>,
     ) -> Result<Self, LlmServiceError>
     where
         M: ModelExecutor + Send + Sync + 'static,
     {
+        let span = info_span!("llm-service");
+        let _span = span.clone();
```

Comment:
is this clone necessary?

Reply:
Yeah, I can't make the code compile without the clone as the `span` gets borrowed. I can't either use 

`let _enter = span.clone().enter();`

As it frees the value while in use.